tests/integration/data/test-bundle-ceph.yaml

```diff
@@ -35,8 +35,12 @@ applications:
     constraints: cores=2 mem=4G root-disk=16G
     num_units: 1
     storage:
-      osd-devices: 1G,2
-      osd-journals: 1G,1
+      osd-devices:
+        count: 2
+        size: '1G'
+      osd-journals:
+        count: 1
+        size: '1G'
```

Comment:
This was is hard-won... but 

[ApplicationDeploy.storage](https://github.com/juju/python-libjuju/blob/master/juju/client/_definitions.py#L1592)
uses [`from_json`](https://github.com/juju/python-libjuju/blob/master/juju/client/facade.py#L663)

where the right side of each key (eg `1G,2`) must either be valid json, or just a `dict` with the correct keys defined in [Constraints](https://github.com/juju/python-libjuju/blob/master/juju/client/_definitions.py#L6511)

Reply:
it seems there's very little to be done here with pylibjuju.  If I go with any deviations to the bundle (like this one above), they fail when python-libjuju uses `fetch_plan` to validate the bundle.

I'm not sure how to proceed with storage constraints from pylibjuju

```yaml
    storage:
      osd-devices: '{count: 2, size: "1G"}'
      osd-journals: '{count: 1, size: "1G"}'
```

fails like this:

```
juju.errors.JujuError: ['invalid storage "osd-devices" in application "ceph-osd": unrecognized storage constraint "{count: 2" not valid', 'invalid storage "osd-journals" in application "ceph-osd": unrecognized storage constraint "{count: 1" not valid']
```

Reply:
the [pylibjuju example](https://github.com/juju/python-libjuju/blob/53cd33d884f3ca710e1fa026c74f31b02281be69/tests/integration/bundle/bundle-with-storage-constraint.yaml) has the original submission -- with a slight deviation

Reply:
but yeah... even the example constraints fails

@cderici take a look here https://github.com/canonical/k8s-operator/actions/runs/9411181657/job/25924049004#step:22:71

Reply:
@addyess yeah this appears like a bug in pylibjuju, and quick look shows that we might have more issues there like resolving things like `ceph-osd` as well, can you open a bug on libjuju so I'll have someone take a closer look?

Reply:
@cderici I raised this issue https://github.com/juju/python-libjuju/issues/1060 in python-libjuju